cmd/prow-job-dispatcher/main.go

```diff
@@ -392,6 +392,9 @@ func main() {
 	if err := dispatchJobs(context.TODO(), o.prowJobConfigDir, o.maxConcurrency, config, jobVolumes); err != nil {
 		logrus.WithError(err).Fatal("Failed to dispatch")
 	}
+	if err := config.AreDeterministic(o.prowJobConfigDir); err != nil {
+		logrus.WithError(err).Fatal("Failed to check if the config can determinize jobs")
+	}
```

Comment:
This looks a bit weird to me, I would expect the dispatcher tool to never generate an incorrect config, not check and fail when it does. Can we detect the situation and fix it instead of failing -- like select one of the options or something else.

Reply:
Removed from dispatcher.
If dispatcher generates a bad config (that would be a bug), then sanitize (presubmit) will failed and the PR from dispatcher wont be merged.